### ISSUE TITLE
In build mode, the compiled CSS file doesn't follow the order of imports described in theme/index.js.

### ISSUE THREAD

**Issue by matheusilva** [vuejs/vitepress#2671]
### Describe the bug

I'm documenting a library of components, and to style the components for this purpose, I use a main stylesheet that is based on Bootstrap. To achieve this, I imported the .css file in index.js, and it works perfectly in dev mode.

However, when I perform the build, the order in which I declared the files in theme/index.js is not respected, and the styles from DefaultTheme are placed first, and only after that, the styles from Bootstrap.css are applied. This ends up breaking the appearance of VitePress, such as <p>, <a>, etc.

### What my config.js is like:

```js
import './bootstrap.css';
import DefaultTheme from 'vitepress/theme';
import './custom.css';

export default{
    ...DefaultTheme,
};
```

### The bootstrap.css:

```css
@charset "UTF-8";
@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap");
:root,
[data-bs-theme=light] {
  --lx-blue: #0574E9;
  --lx-indigo: #6610f2;
  --lx-purple: #805FE3;
  --lx-pink: #d63384;
  --lx-red: #ba161a;
  --lx-orange: #DD6B00;
  --lx-yellow: #FFB200;
  --lx-green: #0BA046;
  --lx-teal: #20c997;
  --lx-cyan: #0dcaf0;
  --lx-black: #000;
  --lx-white: #fff;
  --lx-gray: #626262;
  --lx-gray-dark: #292929;
  --lx-gray-100: #FFF;
  --lx-gray-200: #E1E1E1;
  --lx-gray-300: #BDBDBD;
  --lx-gray-400: #A1A1A1;
  --lx-gray-500: #828282;
  --lx-gray-600: #626262;
  --lx-gray-700: #474747;
  --lx-gray-800: #292929;
  --lx-gray-900: #181818;
  --lx-primary: #411E5A;
  --lx-secondary: #626262;
  --lx-success: #098038;
  --lx-info: #045dba;
  --lx-warning: #FFB200;
  --lx-danger: #ba161a;
  --lx-light: #FFF;
  --lx-dark: #181818;
  [...]
```

### The dist/assets/style.[chunkid].css (unminify):

```css
@charset "UTF-8";@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap");

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-cyrillic.5f2c6c8c.woff2') format('woff2');
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-cyrillic-ext.e75737ce.woff2') format('woff2');
  unicode-range: U+0460-052F, U+1C80-1C88, U+20B4, U+2DE0-2DFF, U+A640-A69F,
    U+FE2E-FE2F;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-greek.d5a6d92a.woff2') format('woff2');
  unicode-range: U+0370-03FF;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-greek-ext.ab0619bc.woff2') format('woff2');
  unicode-range: U+1F00-1FFF;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-latin.2ed14f66.woff2') format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA,
    U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215,
    U+FEFF, U+FFFD;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-latin-ext.0030eebd.woff2') format('woff2');
  unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB,
    U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-vietnamese.14ce25a6.woff2') format('woff2');
  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1,
    U+01AF-01B0, U+1EA0-1EF9, U+20AB;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-cyrillic.ea42a392.woff2') format('woff2');
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-cyrillic-ext.33bd5a8e.woff2') format('woff2');
  unicode-range: U+0460-052F, U+1C80-1C88, U+20B4, U+2DE0-2DFF, U+A640-A69F,
    U+FE2E-FE2F;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-greek.8f4463c4.woff2') format('woff2');
  unicode-range: U+0370-03FF;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-greek-ext.4fbe9427.woff2') format('woff2');
  unicode-range: U+1F00-1FFF;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-latin.bd3b6f56.woff2') format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA,
    U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215,
    U+FEFF, U+FFFD;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-latin-ext.bd8920cc.woff2') format('woff2');
  unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB,
    U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: italic;
  font-named-instance: 'Italic';
  src: url('/assets/inter-italic-vietnamese.6ce511fb.woff2') format('woff2');
  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1,
    U+01AF-01B0, U+1EA0-1EF9, U+20AB;
}

/* Chinese quotes rendering fix. 中英文弯引号共享 Unicode 码位，确保引号使用中文字体渲染 */
@font-face {
  font-family: 'Chinese Quotes';
  src: local('PingFang SC Regular'), local('PingFang SC'), local('SimHei'),
    local('Source Han Sans SC');
  unicode-range: U+2018, U+2019, U+201C, U+201D; /* 分别是 ‘’“” */
}
/**
 * Colors Base
 *
 * These are the pure base color presets. Most of the time, you should not be
 * using these colors directly in the theme but rather use "Colors Theme"
 * instead because those are "Theme (light or dark)" dependant.
 * -------------------------------------------------------------------------- */

:root {
  --vp-c-white: #ffffff;
  --vp-c-black: #000000;

  --vp-c-gray: #8e8e93;
  [...]
```

### Expected behavior

The file `dist/assets/style.[chunkid].css` looked something like this:

```css
@charset "UTF-8";
@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap");
:root,
[data-bs-theme=light] {
  --lx-blue: #0574E9;
  --lx-indigo: #6610f2;
  --lx-purple: #805FE3;
  --lx-pink: #d63384;
  --lx-red: #ba161a;
  --lx-orange: #DD6B00;
  --lx-yellow: #FFB200;
  --lx-green: #0BA046;
  --lx-teal: #20c997;
  --lx-cyan: #0dcaf0;
  --lx-black: #000;
  --lx-white: #fff;
  --lx-gray: #626262;
  --lx-gray-dark: #292929;
  --lx-gray-100: #FFF;
  --lx-gray-200: #E1E1E1;
  --lx-gray-300: #BDBDBD;
  --lx-gray-400: #A1A1A1;
  --lx-gray-500: #828282;
  --lx-gray-600: #626262;
  --lx-gray-700: #474747;
  --lx-gray-800: #292929;
  --lx-gray-900: #181818;
  --lx-primary: #411E5A;
  --lx-secondary: #626262;
  --lx-success: #098038;
  --lx-info: #045dba;
  --lx-warning: #FFB200;
  --lx-danger: #ba161a;
  --lx-light: #FFF;
  --lx-dark: #181818;
  --lx-linx-100: #f9eefb;
  --lx-linx-200: #efddf8;
  --lx-linx-300: #debcf1;
  --lx-linx-400: #c29bda;
  --lx-linx-500: #b388cd;
  --lx-linx-600: #9a6fb4;
  --lx-linx-700: #59366d;
  --lx-linx-800: #411E5A;
  --lx-linx-900: #280A30;
  --lx-silver-000: #F2F4F7;
  --lx-silver-100: #E9ECF0;
  --lx-silver-200: #D0D5DD;
  --lx-silver-300: #A6AEBA;
  --lx-silver-400: #6D7583;
  --lx-silver-500: #4C5462;
  --lx-silver-600: #474747;
  --lx-silver-700: #292929;
  --lx-silver-800: #181818;
  --lx-silver-900: #0E1116;
  --lx-primary-rgb: 65, 30, 90;
  --lx-secondary-rgb: 98, 98, 98;
  --lx-success-rgb: 9, 128, 56;
  --lx-info-rgb: 4, 93, 186;
  --lx-warning-rgb: 255, 178, 0;
  --lx-danger-rgb: 186, 22, 26;
  --lx-light-rgb: 255, 255, 255;
  --lx-dark-rgb: 24, 24, 24;
  }
  .alert {
  --lx-alert-bg: transparent;
  --lx-alert-padding-x: 1rem;
  --lx-alert-padding-y: 1rem;
  --lx-alert-margin-bottom: 1rem;
  --lx-alert-color: inherit;
  --lx-alert-border-color: transparent;
  --lx-alert-border: var(--lx-border-width) solid var(--lx-alert-border-color);
  --lx-alert-border-radius: 0.375rem;
  --lx-alert-link-color: inherit;
  position: relative;
  padding: var(--lx-alert-padding-y) var(--lx-alert-padding-x);
  margin-bottom: var(--lx-alert-margin-bottom);
  color: var(--lx-alert-color);
  background-color: var(--lx-alert-bg);
  border: var(--lx-alert-border);
  border-radius: var(--lx-alert-border-radius);
}

.alert-heading {
  color: inherit;
}

`[...]`

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-cyrillic.5f2c6c8c.woff2') format('woff2');
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}

@font-face {
  font-family: 'Inter var';
  font-weight: 100 900;
  font-display: swap;
  font-style: normal;
  font-named-instance: 'Regular';
  src: url('/assets/inter-roman-cyrillic-ext.e75737ce.woff2') format('woff2');
  unicode-range: U+0460-052F, U+1C80-1C88, U+20B4, U+2DE0-2DFF, U+A640-A69F,
    U+FE2E-FE2F;
}
[...]
/**
 * Colors Base
 *
 * These are the pure base color presets. Most of the time, you should not be
 * using these colors directly in the theme but rather use "Colors Theme"
 * instead because those are "Theme (light or dark)" dependant.
 * -------------------------------------------------------------------------- */

:root {
  --vp-c-white: #ffffff;
  --vp-c-black: #000000;

  --vp-c-gray: #8e8e93;

  --vp-c-text-light-1: rgba(60, 60, 67);
  --vp-c-text-light-2: rgba(60, 60, 67, 0.75);
  --vp-c-text-light-3: rgba(60, 60, 67, 0.33);

  --vp-c-text-dark-1: rgba(255, 255, 245, 0.86);
  --vp-c-text-dark-2: rgba(235, 235, 245, 0.6);
  --vp-c-text-dark-3: rgba(235, 235, 245, 0.38);

  --vp-c-green: #10b981;
  --vp-c-green-light: #34d399;
  --vp-c-green-lighter: #6ee7b7;
  --vp-c-green-dark: #059669;
  --vp-c-green-darker: #047857;
  --vp-c-green-dimm-1: rgba(16, 185, 129, 0.05);
  --vp-c-green-dimm-2: rgba(16, 185, 129, 0.2);
  --vp-c-green-dimm-3: rgba(16, 185, 129, 0.5);

  --vp-c-yellow: #d97706;
  --vp-c-yellow-light: #f59e0b;
  --vp-c-yellow-lighter: #fbbf24;
  --vp-c-yellow-dark: #b45309;
  --vp-c-yellow-darker: #92400e;
  --vp-c-yellow-dimm-1: rgba(234, 179, 8, 0.05);
  --vp-c-yellow-dimm-2: rgba(234, 179, 8, 0.2);
  --vp-c-yellow-dimm-3: rgba(234, 179, 8, 0.5);

  --vp-c-red: #f43f5e;
  --vp-c-red-light: #fb7185;
  --vp-c-red-lighter: #fda4af;
  --vp-c-red-dark: #e11d48;
  --vp-c-red-darker: #be123c;
  --vp-c-red-dimm-1: rgba(244, 63, 94, 0.05);
  --vp-c-red-dimm-2: rgba(244, 63, 94, 0.2);
  --vp-c-red-dimm-3: rgba(244, 63, 94, 0.5);

  --vp-c-sponsor: #db2777;
}
```

### System Info

```sh
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 4.35 GB / 15.39 GB
  Binaries:
    Node: 16.19.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (114.0.1823.82)  
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    vitepress: ^1.0.0-beta.5 => 1.0.0-beta.5
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Duplicate of #2150
